internal/core/metadata/rest_provisionwatcher.go

```diff
@@ -506,16 +506,21 @@ func restDeleteProvisionWatcherById(
 	// Check if the provision watcher exists
 	pw, err := dbClient.GetProvisionWatcherById(id)
 	if err != nil {
-		errorHandler.Handle(w, err, errorconcept.ProvisionWatcher.NotFoundById)
+		errorHandler.HandleOneVariant(
+			w,
+			err,
+			errorconcept.ProvisionWatcher.NotFoundById,
+			errorconcept.Default.InternalServerError)
 		return
 	}
 
-	err = deleteProvisionWatcher(pw, w, lc, dbClient, errorHandler)
-	if err != nil {
+	if err = deleteProvisionWatcher(pw, w, lc, dbClient, errorHandler); err != nil {
 		errorHandler.Handle(w, err, errorconcept.ProvisionWatcher.DeleteError_StatusInternalServer)
 		return
 	}
+
 	w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
+	w.WriteHeader(http.StatusOK)
 	w.Write([]byte("true"))
```

Comment:
I apologize for not mentioning this in my last comment. Can we add an explicit ignore to the returned `error`?

```go
 _ = w.Write([]byte("true")
```

I would really like to start making these little changes to make the code more consistent.